### PR TITLE
Feat/add the faq section to the landing page - MAILPOET-4798

### PR DESCRIPTION
**Pull request by triple0t**
## Description

This PR adds a FAQ to the landing page.

This is a continuation of https://github.com/mailpoet/mailpoet/pull/4626

mailpoet-landingpage FAQ

## Code review notes

Please start reviewing from https://github.com/mailpoet/mailpoet/commit/21292802bd706f75ce6098d7315d4b912d444931

## QA notes

You can access the page here: `wp-admin/admin.php?page=mailpoet-landingpage`

You need to enable Landingpage experimental features here: `/wp-admin/admin.php?page=mailpoet-experimental`

Check https://github.com/mailpoet/mailpoet/pull/4620 for more QA information 

## Linked PRs

* First: https://github.com/mailpoet/mailpoet/pull/4620
* Second: https://github.com/mailpoet/mailpoet/pull/4626
* **Third**: https://github.com/mailpoet/mailpoet/pull/4628

## Linked tickets

[MAILPOET-4798](https://mailpoet.atlassian.net/browse/MAILPOET-4798)

## After-merge notes

**Note**: Please merge after https://github.com/mailpoet/mailpoet/pull/4626

[MAILPOET-4798]: https://mailpoet.atlassian.net/browse/MAILPOET-4798?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/assets/js/src/landingpage/faq.tsx
Comment (samnajian) [inline mailpoet/assets/js/src/landingpage/faq.tsx:20]: Wondering if you've verified `./do translations:build` exports these strings to `mailpoet.pot`?

File: mailpoet/assets/js/src/landingpage/faq.tsx
Comment (samnajian) [inline mailpoet/assets/js/src/landingpage/faq.tsx:69]: I think if these links came from constants we would be able to find and change them easier in future, what do you think?

File: mailpoet/assets/js/src/landingpage/faq.tsx
Comment (samnajian) [inline mailpoet/assets/js/src/landingpage/faq.tsx:6]: Defining these types under a separate named TS `type` would make it easier to re-use and read, what do you think?

File: mailpoet/lib/Config/Changelog.php
Comment (samnajian) [inline mailpoet/lib/Config/Changelog.php]: Redundant usage of `shouldShowLandingPage`, one of them can be removed.

**Review by samnajian — COMMENTED** [review]
feels like this PR has changes from https://github.com/mailpoet/mailpoet/pull/4626/, in this case I think it's better to update this PR and set merge it to 4626 once it's approved, I'll gladly re-review after the update is done.

File: mailpoet/assets/js/src/landingpage/faq.tsx
Comment (triple0t) [reply mailpoet/assets/js/src/landingpage/faq.tsx:20]: Screenshot 2023-01-09 at 12 01 29 PM

Yup, it does.

File: mailpoet/assets/js/src/landingpage/faq.tsx
Comment (triple0t) [reply mailpoet/assets/js/src/landingpage/faq.tsx:69]: Hmmm 🤔, I'm not sure I agree with that.

Taking another look at the file, I think I'm okay with where the links are. 
It's super easy to understand what it refers to, which section and accordion item it describes and how to change it.

I believe it would be more confusing if we have multiple constants for the read-more links. We run into another stress of looking up the constants, checking the values, and understanding the context in which they were used.

File: mailpoet/lib/Config/Changelog.php
Comment (triple0t) [reply mailpoet/lib/Config/Changelog.php]: This method has an early return [here](https://github.com/mailpoet/mailpoet/blob/cd18db8499c2cd2f236f0e01242ac25a2a859c85/mailpoet/lib/Config/Changelog.php#L124-L129)

The landing page is currently an experimental feature. We must ensure the plugin redirects to the welcome wizard pending when the landing page is removed from experimental.

We can refactor the method to remove the duplicate check for `$this->shouldShowLandingPage()` when that is done.

File: mailpoet/assets/js/src/landingpage/faq.tsx
Comment (triple0t) [reply mailpoet/assets/js/src/landingpage/faq.tsx:6]: I don’t believe we will be reusing this type anywhere else. Extracting it into a named `type` adds verbosity to the codebase.

Do let me know if you feel strongly about it and still want me to go ahead and add the type.

**Comment by triple0t** [comment]
> feels like this PR has changes from #4626, in this case I think it's better to update this PR and set merge it to 4626 once it's approved, I'll gladly re-review after the update is done.

Thanks for the review, @samnajian 🙌🏿.

Yes, this PR is based on the work done in #4626.

Yes, this PR will be merged into trunk after #4626.

No, we do not need to merge this PR into #4626.

File: mailpoet/assets/js/src/landingpage/faq.tsx
Comment (samnajian) [reply mailpoet/assets/js/src/landingpage/faq.tsx:6]: It's fine to keep it like this as well.

**Review by samnajian — DISMISSED** [review]
LGTM!

**Review by Aschepikov — APPROVED** [review]
GJ!

**Pull request merged**
